### ISSUE TITLE
Carthage build error due to deployment target mismatch

### ISSUE THREAD

**Issue by pepas-everly** [launchdarkly/ios-client#280]
EDIT: This issue is specifically for Xcode 14.

At some point, one of Launch Darkly's dependencies (`LDSwiftEventSource`) bumped its deployment target to 11.0, but the iOS SDK's deployment target is still set to 10.0:

<img width="1046" alt="Screen Shot 2022-09-12 at 4 04 17 PM" src="https://user-images.githubusercontent.com/99997577/189760010-6fbf0bc6-90ae-4383-a6b9-195ee2d08c3d.png">

This causes Carthage to attempt to build everything using the 10.0 deployment target, which of course causes `LDSwiftEventSource` to fail to build:

```
.../FlagSynchronizer.swift:3:8: error: compiling for iOS 10.0, but module 'LDSwiftEventSource' has a minimum deployment target of iOS 11.0
```

To fix this, please bump the iOS SDK's deployment target to 11.0.

While waiting for this upstream fix, current Carthage users can use the following work-around:

```
carthage update --no-build

# Fix LaunchDarkly by bumping its deployment target.
cd Carthage/Checkouts/ios-client-sdk
    sed -I '' 's/IPHONEOS_DEPLOYMENT_TARGET = 10.0;/IPHONEOS_DEPLOYMENT_TARGET = 11.0;/'  LaunchDarkly.xcodeproj/project.pbxproj
cd - >/dev/null

carthage build --use-xcframeworks --platform iOS --no-use-binaries --cache-builds
```

**Comment by keelerm84** [comment]
Thank you for filing this report. I'm unclear on when this deployment target might have been modified. Looking at the swift eventsource repo, we are targeting iOS v10 as seen [here](https://github.com/launchdarkly/swift-eventsource/blob/main/Package.swift#L8).

I setup a new project locally and was able to build without issue and without a workaround. Are you able to setup a minimal reproducible test case?

**Comment by pepas-everly** [comment]
> I setup a new project locally and was able to build without issue and without a workaround.

Try again with Xcode 14.

You should see this error immediately:

<img width="462" alt="Screen Shot 2022-09-13 at 10 54 00 AM" src="https://user-images.githubusercontent.com/99997577/189949288-61d8fe44-0a99-4a2c-a323-591e8d9ca8ec.png">

Steps to reproduce the above error:
- Unzip https://github.com/launchdarkly/ios-client-sdk/archive/refs/heads/v6.zip
- Hit command + B

**Comment by pepas-everly** [comment]
I just realized I hadn't actually mentioned Xcode 14 anywhere in the original post.  Sorry for the confusion.  I'll update it.

**Comment by pepas-everly** [comment]
> we are targeting iOS v10

Also, 11.0 appears to be the minimum deployment target for Xcode 14.

**Comment by keelerm84** [comment]
We have published v7.0.0. This version drops support for iOS v10, allowing developers to  use Xcode 14. Thank you for your patience while we worked on a fix for this!
